### PR TITLE
Create CRUD operations using MongoDB

### PR DESCRIPTION
**Pull request by Chris-Maina**
#### What does this PR do?
Implements CRUD endpoints using MongoDB. 
#### Description of tasks completed?
Have the following endpoints working: 
* `GET /api/videos`
* `POST /api/videos`
* `GET /api/videos/:slug`
* `PATCH /api/videos/:slug`
* `DELETE /api/videos/:slug`
#### How should this be manually tested?
`git pull` this branch, `cd` into the project folder, `npm install`. Create an `.env` file with `MONGO_URI` and `PORT`  variables filled with your own configs. Use `nodemon index.js` to start the application.
You can now use `Postman` to test out the routes/endpoints.

### CODE REVIEW

**Review by EinsteinNjoroge — APPROVED** [review]
This looks good to me

File: app/models/models.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const videoSchema = new Schema({
+  createdAt: { type: Date, default: Date.now() },
+  modifiedAt: Date,
+  name: { 
+    type: String,
+    default: '',
```
Comment (machariamarigi) [inline app/models/models.js]: This being a string, having a `default` value set to an empty string is redundant when you have `required` set to true and `min-length` set to 3.

File: app/routes.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const Video = require('./models/models');
+
+createSlug = str => (str.split(' ').join('-').toLowerCase());
```
Comment (machariamarigi) [inline app/routes.js]: nice work using slugs 👍

**Review by machariamarigi — COMMENTED** [review]
Nice work in general

**Pull request merged**